Mathlib/Data/List/Basic.lean

```diff
@@ -785,8 +801,31 @@ theorem getLast?_append {l₁ l₂ : List α} {x : α} (h : x ∈ l₂.getLast?)
     exact h
 #align list.last'_append List.getLast?_append
 
+@[simp]
+theorem getLast?_reverse (l : List α) : l.reverse.getLast? = l.head? := by cases l <;> simp
+
+/-! ### getLast! -/
+
+theorem getLast!_eq_getLast_of_ne_nil [Inhabited α] {l : List α} (h : l ≠ []) :
```

Comment:
Do we need `of_ne_nil` in the name?

Reply:
My thought was that if a theorem asks for an assumption, the name should say so. I am quite new to contributing to the Mathlib so I don't know what is the norm here.